### PR TITLE
Remove code related to Beats Central Management

### LINKED ISSUE #76330: Remove code related to Beats Central Management
BCM was removed in 7.14 (https://github.com/elastic/beats/pull/25696, https://github.com/elastic/kibana/pull/99789). There are still leftover references in ES code related to accessing `.management-beats` index and licensing that should be cleaned up.

Please see https://github.com/elastic/elasticsearch/pull/30520, the original PR that introduced BCM-related code, for what to remove.

### PR DESCRIPTION
**Pull request by justincr-elastic**
Removes code originally introduced by #30520

Closes #76330

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SecurityDocumentationIT.java [hunk 1]
```diff
@@ -735,7 +735,7 @@ public void testGetRoles() throws Exception {
             List<Role> roles = response.getRoles();
             assertNotNull(response);
             // 30 system roles plus the three we created
```
Comment (tvernum) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SecurityDocumentationIT.java:737]:
```suggestion
            // 29 system roles plus the three we created
```

**Review by tvernum — CHANGES_REQUESTED** [review]
Unfortunately, removing a builtin role is a breaking change, and we can't do that without a period of deprecation.

We need to keep the `beats_admin` role, but deprecate it.
We can remove the license related code.

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SecurityDocumentationIT.java (hunk 1 above)
Comment (justincr-elastic) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/SecurityDocumentationIT.java]: I fixed the comment. Thank you.

**Comment by justincr-elastic** [comment]
https://www.elastic.co/guide/en/beats/filebeat/8.0/configuration-central-management.html
"Starting in version 7.14, Beats central management has been removed."

@tvernum, since it has been removed from Kibana, is it ok to do ahead and remove it from Elasticsearch 8.x?

**Comment by tvernum** [comment]
No, even though the functionality that it was intended to support has been removed from Kibana, the role itself exists in Elasticsearch and hasn't been marked as deprecated, nor was it ever marked as `beta` (which the Kibana functionality was).

We really do need to deprecate the role instead of removing it.

**Pull request closed without merging**